### PR TITLE
Add ActiveWaiter::EnumerableJob as template

### PR DESCRIPTION
**Pull request by choonkeat**
:eyes: ?

PS: jollygoodstyle check in separate PR

### CODE REVIEW

File: lib/active_waiter/enumerable_job.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    include ActiveWaiter::Job
+  end
+
+  def perform(*args)
+    before(*args)
+    enumerable.each_with_index do |item, index|
+      foreach(item)
+      update_active_waiter(percentage: (100 * (index+1.to_f) / items_count))
+    end
+    after
```
Comment (winston) [inline lib/active_waiter/enumerable_job.rb:14]: Just wondering if `after` should accept `(*args)` too?

**Comment by winston** [comment]
Other than that, looks :heart_eyes:

File: lib/active_waiter/enumerable_job.rb (hunk 1 above)
Comment (choonkeat) [inline lib/active_waiter/enumerable_job.rb:14]: `*args` was given to `before(*args)` earlier as an entry point for `perform(*args)`. if they are needed elsewhere (e.g. `enumerable`, `foreach`, `result`) implementers can store and manage inside `before`

Comment (winston) [inline lib/active_waiter/enumerable_job.rb:14]: :ok_hand:

**Pull request merged**